client/src/redux/sagas/systems.sagas.test.js

```diff
@@ -0,0 +1,45 @@
+import { expectSaga } from 'redux-saga-test-plan';
+import { pushSystemKeys } from 'redux-saga/effects';
+import { fetchUtil } from 'utils/fetchUtil';
+import { vi } from 'vitest';
+
+// Mocks the cross-fetch
+vi.mock('cross-fetch');
+
+// Test the pushSystemKeys function
+describe('pushSystemKeys', () => {
+  it('updates system data'),
+    () => {
+      // Create a mock form for the fetchUtil call
+      const mockForm = {
+        password: 'mockPassword',
+        token: 'mockToken',
+        hostname: 'mockHostname',
+      };
+      expectSaga(pushSystemKeys, {})
+        // Sends the call to update the systems modal
```

Comment:
This doesn't just send the call, it also asserts that the `put` will eventually happen. Meaning, if the `put` doesn't happen (which it should, the test is written correctly), the test will fail because something is wrong in the code.

Reply:
Neat! Good to know! 💪